base/src/main/java/com/gdgistanbul/projectx/android/MainActivity.kt

```diff
@@ -3,27 +3,35 @@ package com.gdgistanbul.projectx.android
 import android.arch.lifecycle.Observer
 import android.arch.lifecycle.ViewModelProviders
 import android.os.Bundle
+import android.support.v4.app.Fragment
 import com.gdgistanbul.projectx.android.base.R
 import com.gdgistanbul.projectx.android.ui.feed.FeedContentViewModel
+import com.gdgistanbul.projectx.android.ui.feed.FeedFragment
 
 class MainActivity : BaseActivity() {
 
-    private lateinit var feedContentViewModel: FeedContentViewModel
+
+    private lateinit var currentFragment: BaseNavigationFragment
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
 
-        feedContentViewModel = ViewModelProviders.of(this, viewModelProviderFactory).get(FeedContentViewModel::class.java)
-
-
-        feedContentViewModel.fetchFeedContent()
+        replaceFragment(FeedFragment())
 
+    }
 
-        feedContentViewModel.feedContentLive.observe(this, Observer {
-            // viewstate.
-        })
+    private fun <F> replaceFragment(fragment: F) where F : Fragment, F : BaseNavigationFragment {
```

Comment:
same point here. should move into an extension file.  I think let' s try to create a method to replace the fragment.  https://proandroiddev.com/kotlin-extensions-to-commit-fragments-safely-de06218a1f4